### PR TITLE
fix installation against python 3.9

### LINKED ISSUE #2974: Installation is broken against python 3.9
### Bug report

**Bug summary**

Can't install the dev version against python 3.9

**Code for reproduction**
```bash
pip install numpy cython
pip install -e .
```
output

```
Obtaining file:///Users/clm/dev/python/yt-project/yt_for_py39
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/clm/.pyenv/versions/3.9.0/envs/yt_39/bin/python3.9 /Users/clm/.pyenv/versions/3.9.0/envs/yt_39/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/tmp2j0_7l6c
       cwd: /Users/clm/dev/python/yt-project/yt_for_py39
  Complete output (63 lines):
  Unable to compile OpenMP test program so Cython
  extensions will be compiled without parallel support
  running egg_info
  creating yt.egg-info
  writing yt.egg-info/PKG-INFO
  writing dependency_links to yt.egg-info/dependency_links.txt
  writing entry points to yt.egg-info/entry_points.txt
  writing requirements to yt.egg-info/requires.txt
  writing top-level names to yt.egg-info/top_level.txt
  writing manifest file 'yt.egg-info/SOURCES.txt'
  Traceback (most recent call last):
    File "/Users/clm/dev/python/yt-project/yt_for_py39/setupext.py", line 309, in finalize_options
      import numpy
  ModuleNotFoundError: No module named 'numpy'

  The above exception was the direct cause of the following exception:

  Traceback (most recent call last):
    File "/Users/clm/.pyenv/versions/3.9.0/envs/yt_39/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
      main()
    File "/Users/clm/.pyenv/versions/3.9.0/envs/yt_39/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/Users/clm/.pyenv/versions/3.9.0/envs/yt_39/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 149, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 130, in _get_build_requires
      self.run_setup()
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 253, in run_setup
      super(_BuildMetaLegacyBackend,
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 145, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 98, in <module>
      setup(
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/Users/clm/.pyenv/versions/3.9.0/lib/python3.9/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/Users/clm/.pyenv/versions/3.9.0/lib/python3.9/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/Users/clm/.pyenv/versions/3.9.0/lib/python3.9/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/command/egg_info.py", line 298, in run
      self.find_sources()
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/command/egg_info.py", line 305, in find_sources
      mm.run()
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/command/egg_info.py", line 536, in run
      self.add_defaults()
    File "/private/var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/pip-build-env-zv0xtyka/overlay/lib/python3.9/site-packages/setuptools/command/egg_info.py", line 572, in add_defaults
      sdist.add_defaults(self)
    File "/Users/clm/.pyenv/versions/3.9.0/lib/python3.9/distutils/command/sdist.py", line 228, in add_defaults
      self._add_defaults_ext()
    File "/Users/clm/.pyenv/versions/3.9.0/lib/python3.9/distutils/command/sdist.py", line 311, in _add_defaults_ext
      build_ext = self.get_finalized_command('build_ext')
    File "/Users/clm/.pyenv/versions/3.9.0/lib/python3.9/distutils/cmd.py", line 299, in get_finalized_command
      cmd_obj.ensure_finalized()
    File "/Users/clm/.pyenv/versions/3.9.0/lib/python3.9/distutils/cmd.py", line 107, in ensure_finalized
      self.finalize_options()
    File "/Users/clm/dev/python/yt-project/yt_for_py39/setupext.py", line 311, in finalize_options
      raise ImportError(
  ImportError: Could not import cython or numpy. Building yt from source requires
      cython and numpy to be installed. Please install these packages using
      the appropriate package manager for your python environment.
  ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/clm/.pyenv/versions/3.9.0/envs/yt_39/bin/python3.9 /Users/clm/.pyenv/versions/3.9.0/envs/yt_39/lib/python3.9/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /var/folders/rd/dkvtm72n4hs20lmd2jsjypcm0000gn/T/tmp2j0_7l6c Check the logs for full command output.
```

I'm 99% sure this is an issue with `pyproject.toml`, I'm hoping to get to a solution shortly.

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

fix #2974

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,8 +7,7 @@ requires = [
   # keep in sync with travis.yml "minimal" specs (Cython and numpy for py36)
   "Cython>=0.26.1",
   "numpy==1.13.3; python_version=='3.6' and platform_system!='AIX'",
-  "numpy==1.18.1; python_version=='3.7' and platform_system!='AIX'",
-  "numpy==1.18.4; python_version=='3.8' and platform_system!='AIX'",
+  "numpy>=1.18.4; python_version>='3.7' and platform_system!='AIX'",
```
Comment (cphyc) [inline pyproject.toml:10]: Looks good to me, but together with #2917 , why don't we add support for Python 3.9 in the CI altogether?

**Review by cphyc — APPROVED** [review]
LGTM.
Note that interestingly, it _seems_ that yt is working faster on Python 3.9 than on Python 3.7.7. More thorough tests would be required to confirm this though.

File: pyproject.toml (hunk 1 above)
Comment (neutrinoceros) [reply pyproject.toml:10]: Of course that's the next step, but I'd rather have the bug fix go in asap while the testing can wait imo :)

Comment (cphyc) [reply pyproject.toml:10]: Fair enough!

**Pull request merged**